django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,87 @@
+from django.core import checks
+from django.db import models
+from django.db.models.expressions import DatabaseDefault
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ("BigSerialField", "SmallSerialField", "SerialField")
+
+
+class SerialFieldMixin:
+    db_returning = True
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("default", DatabaseDefault())
+        super().__init__(
+            *args,
+            **{
+                **kwargs,
+                "blank": True,
+            },
+        )
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        kwargs.pop("blank")
+        kwargs.pop("default")
+        return name, path, args, kwargs
+
+    def check(self, **kwargs):
+        return [
+            *super().check(**kwargs),
+            *self._check_not_null(),
+            *self._check_default(),
+        ]
+
+    def _check_not_null(self):
+        if self.null:
+            return [
+                checks.Error(
+                    "SerialFields do not accept null values.",
+                    obj=self,
+                    id="fields.E910",
+                ),
+            ]
+        else:
+            return []
+
+    def _check_default(self):
+        if self.default != DatabaseDefault():
+            return [
+                checks.Error(
+                    "SerialFields do not accept default values.",
+                    obj=self,
+                    id="fields.E911",
+                ),
+            ]
+        else:
+            return []
```

Comment:
We can check for the `NOT_PROVIDED` singleton here instead of using `DatabaseDefault` for something it wasn't meant for:

```suggestion
        if self.default is NOT_PROVIDED:
            return []
        return [
            checks.Error(
                "SerialFields do not accept default values.",
                obj=self,
                id="fields.E911",
            ),
        ]
```

Reply:
Yes, I agree, this is better.